lib/srv/usermgmt.go

```diff
@@ -389,22 +315,50 @@ func (u *HostUserManagement) UpsertUser(name string, ui *services.HostUsersInfo)
 		}
 
 		return nil
-	})
-	if err != nil {
-		return nil, trace.Wrap(err)
+	}))
+}
+
+// UpsertUser creates a temporary Teleport user in the TeleportServiceGroup
+func (u *HostUserManagement) UpsertUser(name string, ui *services.HostUsersInfo) (io.Closer, error) {
+	// grabbing a copy so we don't mutate the caller's HostUsersInfo
```

Comment:
I think it's probably clearer to express this by accepting a value (rather than a pointer to a value) as opposed to taking a copy.

Reply:
Agreed, I think I copied it initially because I didn't want to mess with anything else that might rely on it being a pointer...but I refactored the function body to only pass around my copy 🙃 Updating!